components/datadog/apps/nginx/k8s.go

```diff
@@ -92,7 +92,8 @@ func K8sAppDefinition(e config.Env, kubeProvider *kubernetes.Provider, namespace
 						"ad.datadoghq.com/nginx.checks": pulumi.String(utils.JSONMustMarshal(
 							map[string]interface{}{
 								"nginx": map[string]interface{}{
-									"init_config": map[string]interface{}{},
+									"init_config":           map[string]interface{}{},
+									"check_tag_cardinality": "high",
```

Comment:
This will have no effect because the checks cardinality is already set to `high` globally in agent’s config: https://github.com/DataDog/test-infra-definitions/blob/d610d4c9aa4be9f5ef0b06b2ec8dd56aa508c055/components/datadog/agent/kubernetes_helm.go#L252

In order to validate the new `check_tag_cardinality` feature, we would need another value, different from the “global” one.